src/core/shape_inference/include/unique_shape_inference.hpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ngraph/validation_util.hpp>
+#include <openvino/op/unique.hpp>
+#include <vector>
+
+namespace ov {
+namespace op {
+namespace v10 {
+
+template <class shape_t>
+void shape_infer(const Unique* op,
+                 const shape_t& input_shape,
+                 std::vector<shape_t>& output_shapes,
+                 std::unique_ptr<int64_t> axis = nullptr) {
```

Comment:
If constant data from the input is needed during shape inference (`axis` value in this case), 
it's usually implemented by constant data map.
For example:
https://github.com/openvinotoolkit/openvino/blob/2ffb91533869a4497aef7037bf0f489ef9af6ce0/src/core/shape_inference/include/gather_shape_inference.hpp#L13-L16